pkg/util/clusterauthorizer/authorizer.go

```diff
@@ -44,36 +48,46 @@ func NewAzRefreshableAuthorizer(log *logrus.Entry, azEnv *azureclient.AROEnviron
 		log:              log,
 		azureEnvironment: azEnv,
 		kubernetescli:    kubernetescli,
-		tokenClient:      tokenClient,
 	}, nil
 }
 
-func (a *azRefreshableAuthorizer) NewRefreshableAuthorizerToken(ctx context.Context) (refreshable.Authorizer, error) {
+func (a *azRefreshableAuthorizer) NewRefreshableAuthorizerToken(ctx context.Context) (autorest.Authorizer, error) {
 	// Grab azure-credentials from secret
 	credentials, err := AzCredentials(ctx, a.kubernetescli)
 	if err != nil {
 		return nil, err
 	}
-	// create service principal token from azure-credentials
-	token, err := a.tokenClient.GetToken(ctx,
-		a.log,
-		string(credentials.ClientID),
-		string(credentials.ClientSecret),
-		string(credentials.TenantID),
-		a.azureEnvironment.ActiveDirectoryEndpoint,
-		a.azureEnvironment.ResourceManagerEndpoint)
+
+	tokenCredential := a.tokenCredential // for unit testing only
```

Comment:
nit: what if, instead of having a field that was only useful during unit tests, we used the field for dependency injection? We'd define an interface, say `TokencredentialGetter` with a `GetTokenCredential` function (returns tokenCredential and error), which in prod used the logic below, but in our tests, was some other implementation? This feels a little cleaner, and doesn't require all the "only useful for tests" stuff, I'm not sure, just a thought.

Reply:
Yeah, this is pretty fugly.  I was trying to quickly fix the conflicts introduced when these unit tests were merged to `master`, and azidentity works significantly differently.  Your interface idea sounds good, I'll see if I can make this cleaner.

Reply:
Implemented something close to your suggestion in a "fixup" commit below.  You're right, it feels cleaner.
There's another instance of this pattern in a subsequent commit, which I'll also rewrite.

Reply:
lgtm!